packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3891,19 +3893,57 @@ export default abstract class SqlIntegration
     }).join("\n")}`;
   }
 
-  getFilterValues(
-    filterIds: string[],
-    filters: FactFilterInterface[]
-  ): string[] {
-    const filterValues: string[] = [];
-    filterIds.forEach((filterId) => {
-      const filter = filters.find((f) => f.id === filterId);
-      if (filter) {
-        filterValues.push(filter.value);
-      }
-    });
-    return filterValues;
+  public getColumnTopValuesQuery({
+    factTable,
+    column,
+    limit = 50,
+  }: ColumnTopValuesParams) {
+    if (column.datatype !== "string") {
+      throw new Error(`Column ${column.column} is not a string column`);
+    }
+
+    const start = new Date();
+    start.setDate(start.getDate() - 7);
+
+    return format(
+      `
+WITH
+  __factTable as (
+    ${compileSqlTemplate(factTable.sql, {
+      startDate: start,
+      templateVariables: {
+        eventName: factTable.eventName,
+      },
+    })}
+  ),
+  __topValues as (
+    SELECT
+      ${column.column} as value,
+      COUNT(*) as count
```

Comment:
nit: I kind of want us to start being a bit more consistent and capitalize keywords. So can we write `AS` here and in the CTE expressions?

I'm personally also using leading commas since so much of what we do is composable and this leads to fewer errors once you get used to it, but I'm less convinced that makes sense since it can be less readable.